components/bl-totop-component/component.json

```diff
@@ -0,0 +1,136 @@
+      "label": "Offset",
+      "showInSettings": true,
+      "defaultValue": "",
+      "hasLogicHandler": false,
+      "handlerId": "offsetLogic",
+      "settingTooltip": "Scroll offset"
+    },
+    {
+      "type": "select",
+      "name": "position",
+      "label": "Position",
+      "showInSettings": true,
+      "defaultValue": "bottom-right",
+      "hasLogicHandler": false,
+      "handlerId": "positionLogic",
+      "options": [
+        {
+          "value": "bottom-right",
+          "label": "bottom-right"
+        },
+        {
+          "value": "bottom-left",
+          "label": "bottom-left"
+        },
+        {
+          "value": "bottom-center",
+          "label": "bottom-center"
+        },
+        {
+          "value": "top-right",
+          "label": "top-right"
+        },
+        {
+          "value": "top-left",
+          "label": "top-left"
+        },
+        {
+          "value": "top-center",
+          "label": "top-center"
+        },
+        {
+          "value": "center-right",
+          "label": "center-right"
+        },
+        {
+          "value": "center-left",
+          "label": "center-left"
+        }
+      ],
+      "settingTooltip": "The position of the up button on the page"
+    },
+    {
+      "type": "color",
+      "name": "backgroundColor",
+      "label": "Background",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "backgroundColorLogic",
+      "settingTooltip": "Up button background"
+    },
+    {
+      "type": "color",
+      "name": "color",
+      "label": "Color",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "colorLogic",
+      "settingTooltip": "The color of the icon inside the up button"
+    },
+    {
+      "type": "text",
+      "name": "size",
+      "label": "Button Size",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "sizeLogic",
+      "defaultValue": "",
+      "settingTooltip": "The size(width/height) of the up button"
+    },
+    {
+      "type": "text",
+      "name": "iconSize",
+      "label": "Icon Size",
+      "settingTooltip": "The size(width/height) of the icon inside the up button",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "iconSizeLogic"
+    },
+    {
+      "type": "text",
+      "name": "indentX",
+      "label": "Indent X",
+      "showInSettings": true,
+      "defaultValue": "",
+      "hasLogicHandler": false,
+      "handlerId": "indentXLogic",
+      "settingTooltip": "Indent of the up button from the edge of the page horizontally"
+    },
+    {
+      "type": "text",
+      "name": "indentY",
+      "label": "Indent Y",
+      "settingTooltip": "Indent of the up button from the edge of the page vertically",
+      "showInSettings": true,
+      "hasLogicHandler": false,
+      "handlerId": "indentYLogic"
+    }
+  ],
+  "eventHandlers": [],
+  "actions": [],
```

Comment:
How about actions 'Hide Scroll Button' and 'Show Scroll Button'

It's not really needed, but - it's training for you to work with actions, and they will not interfere